File: src/main/java/com/climate_monitoring_system/dto/notification/NotificationDTO.java
package com.climate_monitoring_system.dto.notification;

import com.climate_monitoring_system.dto.climatedata.SensorDTO;
import lombok.Data;

import java.sql.Timestamp;

@Data
public class NotificationDTO {
    private long notificationId;
    private NotificationTypeDTO notificationType;
    private SensorDTO sensor;
    private Timestamp timestamp;
    private boolean conditionsSelfResolved;
    private boolean userActionTaken;
    private ActionDTO action;
    private boolean isActive;
}

File: frontend/src/app/dto/notification/NotificationDTO.ts
import {SensorDTO} from "@/app/dto/climatedata/SensorDTO";
import {ActionDTO} from "@/app/dto/notification/ActionDTO";
import {NotificationTypeDTO} from "@/app/dto/notification/NotificationTypeDTO";

export interface NotificationDTO {
    notificationId?: number;
    notificationType: NotificationTypeDTO;
    sensor: SensorDTO;
    timestamp: Date;
    conditionsSelfResolved: boolean;
    userActionTaken: boolean;
    action: ActionDTO | null;
    isActive: boolean;
}